src/Features/LanguageServer/Protocol/Handler/SignatureHelp/SignatureHelpHandler.cs

```diff
@@ -23,19 +23,23 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
     internal class SignatureHelpHandler : AbstractStatelessRequestHandler<LSP.TextDocumentPositionParams, LSP.SignatureHelp?>
     {
         private readonly IEnumerable<Lazy<ISignatureHelpProvider, OrderableLanguageMetadata>> _allProviders;
-
-        public override string Method => LSP.Methods.TextDocumentSignatureHelpName;
-
-        public override bool MutatesSolutionState => false;
-        public override bool RequiresLSPSolution => true;
+        private readonly IGlobalOptionService _globalOptions;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public SignatureHelpHandler([ImportMany] IEnumerable<Lazy<ISignatureHelpProvider, OrderableLanguageMetadata>> allProviders)
+        public SignatureHelpHandler(
+            [ImportMany] IEnumerable<Lazy<ISignatureHelpProvider, OrderableLanguageMetadata>> allProviders,
+            IGlobalOptionService globalOptions)
         {
             _allProviders = allProviders;
+            _globalOptions = globalOptions;
         }
 
+        public override string Method => LSP.Methods.TextDocumentSignatureHelpName;
+
+        public override bool MutatesSolutionState => false;
+        public override bool RequiresLSPSolution => true;
```

Comment:
nit: i do prefer properties above constructors.

Reply:
These are not data. It's not clear what data does an object hold on if properties that do not have backing storage are above constructor.